cloud_functions/climateiq_data_export_cf/main.py

```diff
@@ -211,19 +221,102 @@ def _build_spatialized_model_predictions(
+
     Returns:
-        A Series of h3 indices to the aggregated prediction for the lat/lon center
-        points which fall within each h3 cell.
+        A shapely.Polygon representing the chunk.
     """
-    df["h3"] = df.apply(
-        lambda row: h3.geo_to_h3(row["lat"], row["lon"], H3_LEVEL), axis=1
+    min_x = chunk_metadata["x_ll_corner"]
+    min_y = chunk_metadata["y_ll_corner"]
+    max_x = min_x + chunk_metadata["col_count"] * study_area_metadata["cell_size"]
+    max_y = min_y + chunk_metadata["row_count"] * study_area_metadata["cell_size"]
+
+    boundary_points = [
+        (min_x, min_y),
+        (max_x, min_y),
+        (max_x, max_y),
+        (min_x, max_y),
+        (min_x, min_y),
+    ]
+
+    gdf_src_crs = gpd.GeoDataFrame(
+        geometry=[geometry.Polygon(boundary_points)],
+        crs=study_area_metadata["crs"],
     )
-    # TODO: Finalize which aggregation to calculate for each prediction type.
-    predictions = df.groupby(["h3"]).prediction.agg("max")
-    return predictions
+    gdf_global_crs = gdf_src_crs.to_crs(GLOBAL_CRS)
+    return gdf_global_crs.geometry[0]
+
+
+def _calculate_h3_indexes(
+    study_area_metadata: dict,
+    chunk_metadata: dict,
+    spatialized_predictions: pd.DataFrame,
+) -> pd.Series:
+    """Projects cell centroids to H3 indexes.
+
+    Filters out cells that have projected H3 centroids outside of the current
+    chunk and de-dupes duplicate H3 projections by using the average prediction
+    value from overlapping cells.
+
+    Args:
+        study_area_metadata: A dictionary containing metadata for the study
+        area.
+        chunk_metadata: A dictionary containing metadata for the chunk.
+        spatialized_predictions: A DataFrame containing the lat/lon coordinates of cell
+        centroids in a single chunk along with associated predictions
+
+    Returns:
+        A Series containing H3 indexes in a single chunk along with associated
+        predictions, representing a subset of the full study area results.
+    """
+    # Calculate H3 information for each cell.
+    spatialized_predictions[
+        ["h3_index", "h3_centroid_lat", "h3_centroid_lon", "h3_boundary"]
+    ] = spatialized_predictions.apply(_add_h3_index_details, axis=1)
+
+    # Filter out any cells where the projected H3 centroid falls outside of the
+    # chunk boundary.
+    chunk_boundary = _get_chunk_boundary(study_area_metadata, chunk_metadata)
+    spatialized_predictions = spatialized_predictions[
+        spatialized_predictions.apply(
+            lambda cell: chunk_boundary.contains(
+                geometry.Point(cell["h3_centroid_lon"], cell["h3_centroid_lat"])
+            ),
+            axis=1,
+        )
+    ]
+
+    # TODO: Add logic for fetching predictions from neighboring chunks for boundary
+    # cells.
+    spatialized_predictions = spatialized_predictions.groupby(
+        ["h3_index"]
+    ).prediction.agg("mean")
```

Comment:
FYI there was a discussion with USL about how to handle cells with water level 0 (i.e., where building footprints are) so we will need to refine this aggregation method

Reply:
ack I filed a task to look into this!